src/lib/accessibility/AccessibilityFactory.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  ButtonBehavior,
+  ToggleButtonBehavior,
+  ImageBehavior,
+  InputBehavior,
+  ListBehavior,
+  ListItemBehavior,
+  SelectableListBehavior,
+  SelectableListItemBehavior,
+  MenuItemBehavior,
+  MenuBehavior,
+  DefaultBehavior,
+  TreeBehavior,
+  TreeItemBehavior,
+  GroupBehavior,
+} from './Behaviors/behaviors'
+import { Accessibility } from './interfaces'
+
+export enum AccessibilityType {
+  default,
+  button,
+  toggleButton,
+  image,
+  input,
+  list,
+  listItem,
+  selectableList,
+  selectableListItem,
+  menu,
+  menuItem,
+  tree,
+  treeItem,
+  group,
+}
+
+export class AccessibilityFactory {
+  private static BehaviorsMap: Map<string, Accessibility> = new Map([
```

Comment:
do we really need this map? I mean, essentially, it seems that for a component it is always about consuming desired behavior. More than that, the set of these behaviors is known at time of compilation - thus, why we don't do it much simplier by just exporting each behavior in `index` file of dedicated module? Something like the following

```js
// in accessibility module, index file
...
export { default as ButtonBehavior } from './buttonBehavior.ts'
...
```

and on the client side (pseudo-code):

```js
import { ButtonBehavior } from './behaviors'

class Button extends UIComponent {
   ...
   accessibility = ButtonBehavior
   ...
}
```

With that we will eliminate unnecessary indirection level represented by this map.

Reply:
@kuzhelov this map was created to extend possibility to apply different behaviors to the component. Moreover, to allow end users to create their own behaviors and to pass it as a prop to component they need without changing a component. 

Let's assume you have a button with default ButtonBehavior. End-user wants to have Toggle button behavior with specific key handler inside it and additional aria-* attributes. 
So all he needs is to create behavior derived from IAccessibilityBehavior and add it to map by calling AccessibilityFactory.registerBehavior(name, behavior). Then just to pass behavior through prop using its name.

It could be not the best solution. If you have any other thoughts/ideas on it - let me know! :)